src/typing_extensions.py

```diff
@@ -1135,52 +1205,22 @@ class Point2D(TypedDict):
 
         See PEP 655 for more details on Required and NotRequired.
         """
-        if fields is _marker or fields is None:
-            if fields is _marker:
-                deprecated_thing = "Failing to pass a value for the 'fields' parameter"
-            else:
-                deprecated_thing = "Passing `None` as the 'fields' parameter"
-
-            example = f"`{typename} = TypedDict({typename!r}, {{}})`"
-            deprecation_msg = (
-                f"{deprecated_thing} is deprecated and will be disallowed in "
-                "Python 3.15. To create a TypedDict class with 0 fields "
-                "using the functional syntax, pass an empty dictionary, e.g. "
-            ) + example + "."
-            warnings.warn(deprecation_msg, DeprecationWarning, stacklevel=2)
-            # Support a field called "closed"
-            if closed is not False and closed is not True and closed is not None:
-                kwargs["closed"] = closed
-                closed = None
-            # Or "extra_items"
-            if extra_items is not NoExtraItems:
-                kwargs["extra_items"] = extra_items
-                extra_items = NoExtraItems
-            fields = kwargs
-        elif kwargs:
-            raise TypeError("TypedDict takes either a dict or keyword arguments,"
-                            " but not both")
-        if kwargs:
-            if sys.version_info >= (3, 13):
-                raise TypeError("TypedDict takes no keyword arguments")
-            warnings.warn(
-                "The kwargs-based syntax for TypedDict definitions is deprecated "
-                "in Python 3.11, will be removed in Python 3.13, and may not be "
-                "understood by third-party type checkers.",
-                DeprecationWarning,
-                stacklevel=2,
+        # This runs when creating inline TypedDicts:
+        if not isinstance(args, tuple):
+            args = (args,)
+        if len(args) != 1 or not isinstance(args[0], dict):
+            raise TypeError(
+                "TypedDict[...] should be used with a single dict argument"
             )
```

Comment:
can't this be written more simply as:

```suggestion
        if not isinstance(args, dict):
            raise TypeError(
                "TypedDict[...] should be used with a single dict argument"
            )
```

(and also change the third argument passed to `_create_typeddict` a few lines below from `args[0]` to `args`)

All your tests seem to pass with that change!

Reply:
This is to support passing arguments to `__getitem__()` as a tuple:

```python
TypedDict[({...},)]
```

While it's not that useful in our case as only one argument can be provided, this aligns with other special forms where tuples are supported (see https://github.com/python/cpython/commit/a35a30509820f956d6feeaa0dbf42e9ca82c12bb, where this was done for `Annotated`).

I'll add a test for this.

Reply:
Hrm, I'd be okay with just saying `TypedDict[({"whatever": int},)]` is disallowed, TBH! Especially until/unless we have an actual use-case for it. It seems like a pretty weird thing to be doing. And it's disallowed by other typing special forms that only accept a single argument -- for example:

```pytb
>>> from typing import Optional
>>> Optional[(int,)]
Traceback (most recent call last):
  File "<python-input-1>", line 1, in <module>
    Optional[(int,)]
    ~~~~~~~~^^^^^^^^
  File "/Users/alexw/.pyenv/versions/3.13.1/lib/python3.13/typing.py", line 432, in inner
    return func(*args, **kwds)
  File "/Users/alexw/.pyenv/versions/3.13.1/lib/python3.13/typing.py", line 576, in __getitem__
    return self._getitem(self, parameters)
           ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
  File "/Users/alexw/.pyenv/versions/3.13.1/lib/python3.13/typing.py", line 809, in Optional
    arg = _type_check(parameters, f"{self} requires a single type.")
  File "/Users/alexw/.pyenv/versions/3.13.1/lib/python3.13/typing.py", line 202, in _type_check
    raise TypeError(f"{msg} Got {arg!r:.100}.")
TypeError: typing.Optional requires a single type. Got (<class 'int'>,).
```

Reply:
Ah, wasn't aware that Optional and such did not accept that